src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerSearchViewModel.cs

```diff
@@ -1005,6 +1159,159 @@ internal IEnumerable<PackageManagerSearchElementViewModel> Search(string query)
+                    FuzzyMinSim = LuceneConfig.MinimumSimilarity
+                };
+
+                Query query = parser.Parse(CreateSearchQuery(fnames, searchTerm));
+
+                //indicate we want the first 50 results
+                TopDocs topDocs = Searcher.Search(query, n: LuceneConfig.DefaultResultsCount);
+                for (int i = 0; i < topDocs.ScoreDocs.Length; i++)
+                {
+                    //read back a doc from results
+                    Document resultDoc = Searcher.Doc(topDocs.ScoreDocs[i].Doc);
+
+                    // Get the view model of the package element and add it to the results.
+                    string name = resultDoc.Get(nameof(LuceneConfig.IndexFieldsEnum.Name));
+
+                    var foundPackage = GetViewModelForPackageSearchElement(name);
+                    if (foundPackage != null)
+                    {
+                        packages.Add(foundPackage);
+                    }
+                }
+                return packages;
+            }
+            else
+            {
+                return Search(searchText);
+            }
+        }
+
+        /// <summary>
+        /// To get view model for a package based on its name.
+        /// </summary>s
+        /// <param name="packageName">Name of the package</param>
+        /// <returns></returns>
+        private PackageManagerSearchElementViewModel GetViewModelForPackageSearchElement(string packageName)
+        {
+            var result = PackageManagerClientViewModel.CachedPackageList.Where(e => {
+                if (e.Name.Equals(packageName))
+                {
+                    return true;
+                }
+                return false;
+            });
+
+            if (!result.Any())
+            {
+                return null;
+            }
+
+            return new PackageManagerSearchElementViewModel(result.ElementAt(0), false);
+        }
+
+        /// <summary>
+        /// Creates a search query with adjusted priority, fuzzy logic and wildcards.
+        /// Complete Search term appearing in Name of the package will be given highest priority.
+        /// Then, complete search term appearing in other metadata,
+        /// Then, a part of the search term(if containing multiple words) appearing in Name of the package
+        /// Then, a part of the search term appearing in other metadata of the package.
+        /// Then priority will be given based on fuzzy logic- that is if the complete search term may have been misspelled for upto 2(max edits) characters.
+        /// Then, the same fuzzy logic will be applied to each part of the search term.
+        /// </summary>
+        /// <param name="fields">All fields to be searched in.</param>
+        /// <param name="SearchTerm">Search key to be searched for.</param>
+        /// <returns></returns>
+        private string CreateSearchQuery(string[] fields, string SearchTerm)
```

Comment:
This function is the same with the one in search viewmodel?

Reply:
Yes this is same as if now as I couldn't see any significant difference with other weights. This doesn;t have access to the search viewmodel. So should i move the original method to a common location?

Reply:
Please do

Reply:
Moved this method to LuceneSearchVM